Portal_Perfomance_Employes/Controllers/EmployeeController.cs

```diff
@@ -9,53 +10,57 @@ namespace PortalPerfomanceEmployees.Controllers
     [ApiController]
     public class EmployeeController : ControllerBase
     {
-        private EmployeesDB _employeesDB = new EmployeesDB();
+        private readonly EmployeeContext _context;
 
-        public EmployeeController()
+        public EmployeeController(EmployeeContext context)
         {
-            _employeesDB.PopularEmployeesDB();
+            _context = context;
         }
 
         [HttpGet]
-        public async Task<IEnumerable<Employee>> Get() => _employeesDB.employees.ToList();
+        public async Task<IActionResult> GetEmployees()
+        {
+            return Ok(await _context.Employees.ToListAsync());
+        }
         
         [HttpGet("{id}")]
-        public async Task<IActionResult> GetbyId(int id)
+        public async Task<IActionResult> GetEmployee(int id)
         {
-
-            Employee emp = _employeesDB.employees.Find(x => x.Id == id);
-
-            return emp == null ? NotFound() : Ok(emp) ;
+            var emp = await _context.Employees
+                .FirstOrDefaultAsync(e => e.Id == id);
+            return emp == null ? NotFound("Employee with that ID was not found") : Ok(emp) ;
         }
 
         [HttpPost]
-        public async Task<IEnumerable<Employee>> CreateEmployee(Employee employee)
+        public async Task<IActionResult> CreateEmployee(Employee employee)
         {
-            _employeesDB.employees.Add(employee);
```

Comment:
I encountered a problem that if you change the id from zero to any number in the POST request, it causes an Error: response status is 500

I solved the problem by adding `employee.Id = 0;` before `_context.Employees.Add(employee);`
![image](https://user-images.githubusercontent.com/108728294/185225009-64a4b00f-2a44-4d9c-833b-0791fba27c9b.png)


And the result before add that code: 
![image](https://user-images.githubusercontent.com/108728294/185225287-ad7c44e7-5bab-479e-8af5-83835926e905.png)

And After: 
![image](https://user-images.githubusercontent.com/108728294/185225458-7fd33d9a-4438-437f-99cc-7118a1b71b18.png)

Reply:
(hopefully) resolved by introducing an `EmployeeDTO` as the POST method body. The `Id` and `Created` fields should no longer be required and should automatically be assigned on the backend

Reply:
(hopefully) resolved by introducing an EmployeeDTO as the POST method body. The Id and Created fields should no longer be required and should automatically be assigned on the backend